### DISCUSSION TITLE
Capture output with pytest

### DISCUSSION THREAD

**Discussion by ConstantinoSchillebeeckx** [fastapi/typer#714] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from rich import print
from typing_extensions import Annotated

app = typer.Typer(add_completion=False, no_args_is_help=True, rich_markup_mode="rich")


@app.command()
def hello(name: str):
    print(f"Hello {name}")


if __name__ == "__main__":
    typer.run(hello)
```

### Description

How do I capture the output of the CLI execution when executing with pytest?

For example, the following unit test fails:
```
from typer.testing import CliRunner

from foo import app

runner = CliRunner()

def test_app():
    result = runner.invoke(app)
    assert result.exit_code == 1
```
The output looks like:
<img width="1030" alt="image" src="https://github.com/tiangolo/typer/assets/8518288/efe09475-14ea-4208-a948-340c0e343e2e">

I would have expected some sort of captured output like
> Missing argument 'NAME'.

This [issue](https://github.com/tiangolo/typer/issues/196) seems to suggest it should just work?

### Operating System

macOS

### Typer Version

0.9.0

### Python Version

3.11.6

**Comment by The-Compiler** [comment, marked as answer]
The runner does its own output capturing: https://typer.tiangolo.com/tutorial/testing/#check-the-result - unfortunately there seems to be no way to turn it off, so you'd need to `print(result.stdout)` at some point (possibly via a helper function or fixture).

Also see (the `CliRunner` is just a thin wrapper over Click):

- https://github.com/pallets/click/issues/737
